### ISSUE TITLE
Highlight groups in syntax.txt are not highlighted on Windows

### ISSUE THREAD

**Issue by gdupras** [vim/vim#10397]
### Steps to reproduce

1. In cmd.exe do `gvim --clean` or `vim --clean`.
2. `:help hl-ColorColumn`
3. The highlight groups are not highlighted:
![hl-bug-windows](https://user-images.githubusercontent.com/57042631/167632542-88aa5308-e069-49e8-9133-31644de6e6ba.png)

### Expected behaviour

The highlight groups should be highlighted, like on Linux:
![Screenshot_20220509_223411](https://user-images.githubusercontent.com/57042631/167632633-2f84a83a-dd1e-42e8-ab86-a8c20257825a.png)

### Version of Vim

8.2.4929 (from vim/vim-win32-installer/releases)

### Environment

Windows 10
cmd.exe

**Comment by lacygoill** [comment]
If you run these Ex commands:

    :help hl-ColorColumn
    /^ColorColumn\t

You cursor should end up on the name of the `ColorColumn` group which is meant to be highlighted.
From this position, what is the output of these Ex commands:

    :echo prop_type_list({'bufnr': bufnr('%')})->index('help-hl-ColorColumn')
    :echo prop_list(line('.'))

**Comment by gdupras** [comment]
This is what I get:
```
:echo prop_type_list({'bufnr': bufnr('%')})->index('help-hl-ColorColumn')
-1
:echo prop_type_list({'bufnr': bufnr('%')})
[]
:echo prop_list(line('.'))
[]
```

**Comment by lacygoill** [comment]
If you write this code in a script:
```vim
vim9script
help syntax
search("^ErrorMsg\t")
var word: string = 'ErrorMsg'
var buf: number = bufnr('%')
prop_type_add('help-hl-' .. word, {
    bufnr: buf,
    highlight: word,
    combine: false,
})
prop_add(line('.'), 1, {length: word->strlen(), type: 'help-hl-' .. word})
```
Then, start Vim with no config and source the previous script:

    vim -Nu NONE -S /path/to/script

Is `ErrorMsg` highlighted as an error?

**Comment by lacygoill** [comment]
Also, for this to work, Vim needs to have been compiled with the `+textprop` feature.  Is `+textprop` present in the output of `:version`?

**Comment by gdupras** [comment]
> Is ErrorMsg highlighted as an error?

Yes:
![temp1](https://user-images.githubusercontent.com/57042631/167660471-435703f0-c8d8-42b6-87b8-dc38e70c78b6.png)

>  Is +textprop present in the output of :version?

Yes:
![temp2](https://user-images.githubusercontent.com/57042631/167660832-d18e8ce6-ddab-49a7-8c4b-85aa2a93bee0.png)

@LemonBoy Yes, changing that line as you suggested fixed the problem.

**Comment by gdupras** [comment]
I get this error:
```
"syntax.txt" [readonly][unix] 5824L, 231142B
Error detected while processing BufRead Autocommands for "C:/Program Files/Vim/vim82/doc/*.txt"..FileType Autocommands for "*"..Syntax Autocommands for "*"..function <SNR>8_SynSet[25]..script C:\Program Files\Vim\vim82\syntax\help.vim:
line  218:
E492: Not an editor command: {
line  219:
E1124: ":var" cannot be used in legacy Vim script
line  221:
E492: Not an editor command:     path = path->tr('/', '\')
line  223:
E121: Undefined variable: path
line  228:
E1128: } without {: }
```

**Comment by lacygoill** [comment]
OK, new try:
```diff
diff --git a/runtime/syntax/help.vim b/runtime/syntax/help.vim
index f5a50bfac..d6e526f6e 100644
--- a/runtime/syntax/help.vim
+++ b/runtime/syntax/help.vim
@@ -215,12 +215,17 @@ hi def link helpError		Error
 hi def link helpTodo		Todo
 hi def link helpURL		String
 
-if expand('%:p') == $VIMRUNTIME .. '/doc/syntax.txt'
+let s:path = $VIMRUNTIME .. '/doc/syntax.txt'
+if exists('+shellslash') && !&shellslash
+  let s:path = s:path->tr('/', '\')
+endif
+if expand('%:p') == s:path
   " highlight groups with their respective color
   import 'dist/vimhelp.vim'
   call vimhelp.HighlightGroups()
 endif
 
+
 let b:current_syntax = "help"
 
 let &cpo = s:cpo_save
```

**Comment by gdupras** [comment]
Yeah, this worked on Windows. I think it should work on Linux too but I cannot test it at the moment.

**Comment by brammool** [comment]
Does this work:
```
if expand('%:p') =~ $VIMRUNTIME .. '[/\\]doc[/\\]syntax.txt'
```

**Comment by gdupras** [comment]
This does not work on Windows. I think it has to do with backslashes.

Here's a sample session that could help figure out why it's not working.

```
:echo expand('%:p') =~ $VIMRUNTIME .. '[/\\]doc[/\\]syntax.txt'
0

:echo expand('%:p')
C:\Program Files\Vim\vim82\doc\syntax.txt

:echo $VIMRUNTIME
C:\Program Files\Vim\vim82

:echo '[/\\]doc[/\\]syntax.txt'
[/\\]doc[/\\]syntax.txt

:echo  $VIMRUNTIME .. '[/\\]doc[/\\]syntax.txt'
C:\Program Files\Vim\vim82[/\\]doc[/\\]syntax.txt

:echo 'C:\Program Files\Vim\vim82\doc\syntax.txt' =~ $VIMRUNTIME .. '[/\\]doc[/\\]syntax.txt'
0

:echo 'C:\Program Files\Vim\vim82\doc\syntax.txt' =~ 'C:\Program Files\Vim\vim82\doc\syntax.txt'
0

echo '\doc\syntax.txt'  =~ '[/\\]doc[/\\]syntax.txt'
1

:echo expand('%:p') =~ '[/\\]doc[/\\]syntax.txt'
1
```

**Comment by lacygoill** [comment]
FWIW, the last posted patch works on Windows:

   > Yeah, this worked on Windows.

And I've tested it on Linux where it works too.

**Comment by gdupras** [comment]
Yeah, your patch does work. I like @brammool's one-liner though and I don't understand why it doesn't work.

Is below the expected behaviour?
```
:echo 'foo\bar' =~ 'foo\bar'
0
```
I would expect the output to be "1". These are single quoted string and `:help literal-string` says
```
This string is taken as it is.  No backslashes are removed or have a special
meaning.  The only exception is that two quotes stand for one quote.

Single quoted strings are useful for patterns, so that backslashes do not need
to be doubled.
```
Something is weird with those backslashes or I'm missing something.

**Comment by brammool** [comment]
Right, the backslashes in $VIMRUNTIME need to be escaped.
Would expand() use forward slashes while $VIMRUNTIME has backward slashes?

**Comment by gdupras** [comment]
The latest runtime files update (3f32a5f1) partially fixes the issue with `syntax/help.vim`.

There is still a problem on Windows when `shellslash` is set. To reproduce:
```
vim --clean -c "set shellslash -c "help hl-ColorColumn"
``` 
And the highlight groups are not highlighted.

This is because `expand('%:p')` contains slashes but `$VIMRUNTIME` contains backslashes in this line:
https://github.com/vim/vim/blob/d881d1598467d88808bafd2fa86982ebbca7dcc1/runtime/syntax/help.vim#L218

**Comment by gdupras** [comment]
Fixed in the latest runtime files update (30ab04e16e1e). Thanks!

Closing.
